bash_completionsV2.go

```diff
@@ -233,7 +233,14 @@ __%[1]s_handle_standard_completion_case() {
 
     # Short circuit to optimize if we don't have descriptions
     if [[ "${completions[*]}" != *$tab* ]]; then
-        IFS=$'\n' read -ra COMPREPLY -d '' < <(compgen -W "${completions[*]}" -- "$cur")
+        local compgen_words=$(printf "%%s\n" "${completions[@]}")
+        IFS=$'\n' read -ra COMPREPLY -d '' < <(IFS=$'\n' compgen -W "${compgen_words}" -- "$cur")
```

Comment:
If we want to support other bash special characters, we need more quoting:

```bash
local compgen_words="$(printf "%%q\n" "${completions[@]}")"
cur="$(printf "%%q" "${cur}")"
IFS=$'\n' read -ra COMPREPLY -d '' < <(IFS=$'\n' compgen -W "${compgen_words}" -- "${cur}")
```

`compgen`'s `-W` honors shell quoting, so a completion like `foo\>bar` would become `foo>bar` to compgen. We also have to quote `${cur}`  because `compgen` appears to honor shell quoting _after_ checking to make sure each word begins with `${cur}`

----

Similarly, we need quoting up above: https://github.com/spf13/cobra/blob/main/bash_completionsV2.go#L62-L84

```bash
$ args=( echo "foo\>bar" )
$ eval "${args[*]}"
foo>bar
```

`eval` also honors shell quoting, so if we want to pass the actual quoted value through to the `__complete` call, we need to quote the args in `requestComp`

```bash
$ args=( echo "foo\>bar" )
$ eval "$(printf "%q " "${args[@]}")"
foo\>bar
```